ecosystem_tests/dorkl.py

```diff
@@ -765,3 +794,385 @@ def basic_blueprint_test(blueprint_file_name,
         except Exception as e:
             logger.error('Error: {e}'.format(e=str(e)))
             cleanup_on_failure(test_name)
+
+
+def is_first_invocation(test_name):
+    """
+    Check if this is the first invocation of the test,
+    by check existence of blueprint and deployment with test_name id.
+    param: test_name: The test name.
+    """
+    logger.info(
+        'Checking if {test_name} in blueprints or deployments list '.format(
+            test_name=test_name))
+    blueprints_list = cloudify_exec('cfy blueprints list')
+    deployments_list = cloudify_exec('cfy deployments list')
+
+    def _map_func(bl_or_dep_dict):
+        return bl_or_dep_dict["id"]
+
+    if test_name in map(_map_func, blueprints_list) and test_name in map(
+            _map_func, deployments_list):
+        logger.info('Not first invocation!')
+        return False
+    elif test_name in map(_map_func, blueprints_list) or test_name in map(
+            _map_func, deployments_list):
+        raise EcosystemTestException(
+            'Found blueprint or deployment with name:{test_name}, please '
+            'delete the blueprint/deployment manually and rerun the '
+            'test'.format(
+                test_name=test_name))
+    else:
+        logger.info('First invocation!')
+        return True
```

Comment:
A deployment cannot exist if the blueprint does not exist. So you could say, if deployment exist, then not first. We don't really care if there is a previous blueprint, because that is actually valid. For example, if we will run deployment update, the previous blueprint will already exist.